File: Exercism/typescript/roman-numerals/roman-numerals.ts
export const toRoman = (_n: number): string => {
	const values: {[key: string]: number} = {
		M: 1000, CM: 900, D: 500, CD: 400, C: 100, XC: 90, L: 50, XL: 40,
		X: 10, IX: 9, V: 5, IV: 4, I: 1
	}

	const orders: string[] = [
		'M', 'CM', 'D', 'CD', 'C', 'XC', 'L', 'XL', 'X', 'IX', 'V', 'IV', 'I'
	]

	let roman = ''

	for(const symbol of orders) {
		while(_n >= values[symbol]) {
			roman += symbol
			_n -= values[symbol]
		}
	}

	return roman
}

File: Exercism/cpp/roman-numerals/roman_numerals.cpp
#include "roman_numerals.h"

namespace roman_numerals {
	std::string convert(int _n)
	{
		std::map<std::string, int> values{
			{"M", 1000}, {"CM", 900}, {"D", 500}, {"CD", 400}, {"C", 100}, {"XC", 90},
			{"L", 50}, {"XL", 40}, {"X", 10}, {"IX", 9}, {"V", 5}, {"IV", 4}, {"I", 1}
		};

		const std::vector<std::string> orders{
			"M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"
		};

		std::string roman = "";

		for(const auto& c : orders)
		{
			while(_n >= values[c])
			{
				roman += c;
				_n -= values[c];
			};
		}

		return roman;
	}
}  // namespace roman_numerals
